tests/e2e/utils/infrastructure.py

```diff
@@ -138,25 +144,86 @@ def create_lambda_functions(self, function_props: Optional[Dict] = None) -> Dict
         return output
 
     def deploy(self) -> Dict[str, str]:
-        """Creates CloudFormation Stack and return stack outputs as dict
+        """Synthesize and deploy a CDK app, and return its stack outputs
+
+        NOTE: It auto-generates a temporary CDK app to benefit from CDK CLI lookup features
 
         Returns
         -------
         Dict[str, str]
             CloudFormation Stack Outputs with output key and value
         """
-        template, asset_manifest_file = self._synthesize()
-        assets = Assets(asset_manifest=asset_manifest_file, account_id=self.account_id, region=self.region)
-        assets.upload()
-        self.stack_outputs = self._deploy_stack(self.stack_name, template)
-        return self.stack_outputs
+        stack_file = self._create_temp_cdk_app()
+        synth_command = f"npx cdk synth --app 'python {stack_file}' -o {self._cdk_out_dir}"
+        deploy_command = (
+            f"npx cdk deploy --app '{self._cdk_out_dir}' -O {self._stack_outputs_file} --require-approval=never"
+        )
```

Comment:
I'm under the impression that since we're using `npx` here, we might not have to use `package.json`. While I like that idea, this makes me wonder how we can control new CDK releases not affecting our E2E silently.

Reply:
It's the contrary, we're using `npx` to use a local CDK CLI installation to make it reliable. The way `npx` works is it'll try to find CDK CLI in `./node_modules/.bin` first, then lookup for a global CDK CLI, and if not found it'll install the latest version.

https://www.npmjs.com/package/npx